navigator.scheduling.isInputPending (#27381)
mdn/content: files/en-us/web/api/scheduling/isinputpending/index.md

---
title: "Scheduling: isInputPending() method"
short-title: isInputPending()
slug: Web/API/Scheduling/isInputPending
page-type: web-api-instance-method
browser-compat: api.Scheduling.isInputPending
---

> `navigator.scheduling.isInputPending` is an experimental feature that is part of the JavaScript Scheduling API. It allows you to check whether there are pending input events in the event queue, indicating that the user has interacted with the page or the system is about to dispatch such events.

Reviewer: ```suggestion
The `isInputPending()` method allows you to check whether there are pending input events in the event queue, indicating that the user has interacted with the page or the system is about to dispatch such events.
```

---
title: "Scheduling: isInputPending() method"
short-title: isInputPending()
slug: Web/API/Scheduling/isInputPending
page-type: web-api-instance-method
browser-compat: api.Scheduling.isInputPending
---

{{SeeCompatTable}}{{APIRef("Scheduling API")}}

The `isInputPending()` method allows you to check whether there are pending input events in the event queue, indicating that the user has interacted with the page or the system is about to dispatch such events.

This feature can be useful in scenarios where you want to check if there are any pending input events before performing a long-running operation or updating the UI. By checking the value of navigator.scheduling.isInputPending, you can avoid blocking the main thread when the user is interacting with the page.

## Syntax

```js-nolint
navigator.scheduling.isInputPending(options)
```

### Parameters

- `options`

  - : An object containing options.
    - `includeContinuous`: A boolean (optional), indicate that continuous input events should also be considered when checking for pending input.

## Value

A boolean.

> The `navigator.scheduling.isInputPending` property returns a Boolean value, indicating whether there are pending input events or not. It can have one of the following three possible values:

Reviewer: ```suggestion
The `navigator.scheduling.isInputPending()` method returns a Boolean value, indicating whether there are pending input events or not. It can have one of the following three possible values:
```

---
title: "Scheduling: isInputPending() method"
short-title: isInputPending()
slug: Web/API/Scheduling/isInputPending
page-type: web-api-instance-method
browser-compat: api.Scheduling.isInputPending
---

{{SeeCompatTable}}{{APIRef("Scheduling API")}}

The `isInputPending()` method allows you to check whether there are pending input events in the event queue, indicating that the user has interacted with the page or the system is about to dispatch such events.

This method can be useful in scenarios where you want to check if there are any pending input events before performing a long-running operation or updating the UI. By checking the value of `navigator.scheduling.isInputPending`, you can avoid blocking the main thread when the user is interacting with the page.

> The property is read-only.

Reviewer: I think this line should be deleted.

Author: Deleted
